src/util/include/util/Interval.h

```diff
@@ -83,8 +83,8 @@ class Interval {
      * @return true if *this and other intersect, false otherwise
      */
     [[maybe_unused]] std::optional<Interval<T>> intersect(const Interval& other) const {
-        double newMin = std::max(min, other.min);
-        double newMax = std::min(max, other.max);
+        T newMin = std::max(min, other.min);
```

Comment:
@bhennion what is the background, that this interval is left open and not right open? This is unintuitive in computer science, because ranges are right-open, but intervals are not.

Reply:
Yes, I mean, that an integral interval class should model [a,b). I don't like complete closed intervals, because they need special treatment for integers (length) and they can't model empty ranges.